DQMOffline/L1Trigger/src/L1TTauOffline.cc

```diff
@@ -533,10 +533,21 @@ bool L1TTauOffline::matchHlt(edm::Handle<trigger::TriggerEvent> const& triggerEv
 
   trigger::TriggerObjectCollection trigObjs = triggerEvent->getObjects();
 
-  for (auto trigIndexIt = m_trigIndices.begin(); trigIndexIt != m_trigIndices.end(); ++trigIndexIt) {
+  for (vector<int>::const_iterator trigIndexIt = m_trigIndices.begin(); trigIndexIt != m_trigIndices.end();
+       ++trigIndexIt) {
     const vector<string> moduleLabels(m_hltConfig.moduleLabels(*trigIndexIt));
-    const unsigned moduleIndex = m_hltConfig.size((*trigIndexIt)) - 2;
-
+    // V.M. 2023.03.08 Same issue as in the L1TMuonDQMOffline.cc - some modules are behind hltBoolEnd, but we want the last one before the "hltBoolEnd"
+    unsigned int moduleIndex = 999999;
+    for (int ii = 0; ii < int(moduleLabels.size()); ii++) {
+      if (moduleLabels[ii] == "hltBoolEnd") {
+        moduleIndex = ii - 1;
```

Comment:
This gets in trouble if `ii = 0`
Probably the loop should start from `ii=1` (see above, where the loop was further proposed in reverse order to speed up, it being more probable that the hltBoolEnd is at the end of the HLT path rather than at the beginning)